src/ansys/simai/core/data/projects.py

```diff
@@ -205,6 +205,16 @@ def compute_gc_formula(
 
         return gc_compute.result if gc_compute.is_ready else None
 
+    def cancel_training(self):
+        """Cancel training if there is one in progress."""
+
+        self.reload()
+        if self.fields.get("is_being_trained") is False:
+            return False
```

Comment:
Two things:
- I'm not sure we need to reload here, we could probably use `self.is_pending()`.
- Why do we return `False` ? This isn't documented in the types or the description and it doesn't make much sense. IMO this fucking should always return `None`. Arguably we could raise an error if the project/model isn't cancellable (if we don't already, not sure how the endpoint behaves).

So basically, I think this whole function could just be `self._client._api.cancel_build(self.id)`

Reply:
The endpoint responds 200 regardless of build status, so raising a `ProcessingError` makes sense.
Using `is_pending`: seems to return `True` for every project object..? (the model object would be the one pending during build). Thus, sticking to using `is_being_trained` field on the project; reload is indeed necessary (according to thorough live testing on dev1).